Source/Core/Common/Config/Layer.cpp

```diff
@@ -38,64 +75,50 @@ Layer::~Layer()
   Save();
 }
 
-bool Layer::Exists(System system, const std::string& section_name, const std::string& key)
+bool Layer::Exists(const ConfigLocation& location) const
 {
-  Section* section = GetSection(system, section_name);
-  if (!section)
-    return false;
-  return section->Exists(key);
+  const auto iter = m_map.find(location);
+  return iter != m_map.end() || iter->second.has_value();
 }
 
-bool Layer::DeleteKey(System system, const std::string& section_name, const std::string& key)
+bool Layer::DeleteKey(const ConfigLocation& location)
 {
-  Section* section = GetSection(system, section_name);
-  if (!section)
-    return false;
-  return section->Delete(key);
+  m_is_dirty = true;
+  bool had_value = m_map[location].has_value();
+  m_map[location].reset();
+  return had_value;
 }
 
-Section* Layer::GetSection(System system, const std::string& section_name)
+void Layer::DeleteAllKeys()
 {
-  for (auto& section : m_sections[system])
-    if (!strcasecmp(section->m_name.c_str(), section_name.c_str()))
-      return section.get();
-  return nullptr;
+  m_is_dirty = true;
+  for (auto& pair : m_map)
+  {
+    pair.second.reset();
+  }
 }
 
-Section* Layer::GetOrCreateSection(System system, const std::string& section_name)
+Section Layer::GetSection(System system, const std::string& section)
```

Comment:
This could also reasonably have a `const` variant as well couldn't it (since it doesn't modify the instance itself anymore)?

Reply:
Pushed something in a012624572.